src/Xamarin.Android.Build.Tasks/Tasks/LinkAssemblies.cs

```diff
@@ -69,24 +69,6 @@ IEnumerable<AssemblyDefinition> GetRetainAssemblies (DirectoryAssemblyResolver r
 
 		public override bool Execute ()
 		{
-			Log.LogDebugMessage ("LinkAssemblies Task");
-			Log.LogDebugMessage ("  UseSharedRuntime: {0}", UseSharedRuntime);
-			Log.LogDebugMessage ("  MainAssembly: {0}", MainAssembly);
-			Log.LogDebugMessage ("  OutputDirectory: {0}", OutputDirectory);
-			Log.LogDebugMessage ("  OptionalDestinationDirectory: {0}", OptionalDestinationDirectory);
-			Log.LogDebugMessage ("  I18nAssemblies: {0}", I18nAssemblies);
-			Log.LogDebugMessage ("  LinkMode: {0}", LinkMode);
-			Log.LogDebugMessage ("  LinkSkip: {0}", LinkSkip);
-			Log.LogDebugTaskItems ("  LinkDescriptions:", LinkDescriptions);
-			Log.LogDebugTaskItems ("  ResolvedAssemblies:", ResolvedAssemblies);
-			Log.LogDebugMessage ("  EnableProguard: {0}", EnableProguard);
-			Log.LogDebugMessage ("  ProguardConfiguration: {0}", ProguardConfiguration);
-			Log.LogDebugMessage ("  DumpDependencies: {0}", DumpDependencies);
-			Log.LogDebugMessage ("  LinkOnlyNewerThan: {0}", LinkOnlyNewerThan);
-			Log.LogDebugMessage ("  HttpClientHandlerType: {0}", HttpClientHandlerType);
-			Log.LogDebugMessage ("  TlsProvider: {0}", TlsProvider);
-			Log.LogDebugMessage ("  PreserveJniMarshalMethods: {0}", PreserveJniMarshalMethods);
-
 			var rp = new ReaderParameters {
 				InMemory    = true,
```

Comment:
For future explorations, we need to see if this can be removed.

Reply:
I tried this just now and it seems like it slows things down:
```
23606 ms  LinkAssemblies                             1 calls
```

So I'm guessing that `InMemory` makes things way faster if you are going to do alot of work looping over every `TypeDefinition`, etc. It must read from the file as you go in some cases?

Reply:
Interesting. Now that you reminded me this feature, I will try to use it in the *jnimarshalmethod-gen*.

Reply:
I think `InMemory` can be bad sometimes too. Like in this case: https://github.com/xamarin/xamarin-android/pull/2148/files

The code was only looking for one assembly attribute and returning true/false, so it would be bad to use it there.